File: client/lib/schema/container_template.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'container_template.freezed.dart';
part 'container_template.g.dart';

@freezed
class ContainerTemplate with _$ContainerTemplate {
  const factory ContainerTemplate({
    required String id,
    required String label,
    @JsonKey(name: "image_url") required String imageUrl,
    @JsonKey(name: "container_weight_gram") required double containerWeightGram,
    @JsonKey(name: "container_max_weight_gram") required double containerMaxWeightGram,
  }) = _ContainerTemplate;

  factory ContainerTemplate.fromJson(Map<String, dynamic> json) => _$ContainerTemplateFromJson(json);
}

File: core-api/pkg/handler/schema/container_template.go
package schema

import "github.com/gin-gonic/gin"

type ContainerTemplate struct {
	Id                     string  `json:"id"`
	Label                  string  `json:"label"`
	ImageUrl               string  `json:"image_url"`
	ContainerWeightGram    float32 `json:"container_weight_gram"`
	ContainerMaxWeightGram float32 `json:"container_max_weight_gram"`
}

type ContainerTemplateController interface {
	GetContainerTemplates(c *gin.Context)
}
